src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -162,6 +146,30 @@ private IEventReferenceExpression CreateBoundEventAccessOperation(BoundEventAssi
                  invokedAsExtensionMethod: invokedAsExtensionMethod);
         }
 
+        private IInvalidExpression CreateInvalidExpressionForHasArgumentsExpression(BoundNode ReceiverOpt, ImmutableArray<BoundExpression> arguments, BoundExpression AdditionalNodeOpt, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit)
+        {
+            Lazy<ImmutableArray<IOperation>> children = new Lazy<ImmutableArray<IOperation>>(
+                      () =>
+                      {
+                          ArrayBuilder<IOperation> builder = ArrayBuilder<IOperation>.GetInstance();
+
+                          if (ReceiverOpt != null && !ReceiverOpt.WasCompilerGenerated)
```

Comment:
What is the scenario where the receiver is compiler generated but should be dropped?

Reply:
For example, in the code below:
```cs
class P
{
    void M1()
    {
        M2(1, 2);
    }

    void M2(int x = 0)
    { }
}
```
There would a compiler generated `BoundThis` for the invalid call `M2(1, 2)`.

Reply:
Why drop the implicit `BoundThis`? The `this` receiver seems correct in that case.

Reply:
I think the reason is we are creating an InvalidExpression instead of an InvocationExpression for example, which basically just a list of operation in user's code. In this case, including a compiler generated `this` can be confusing and not much user can do with it.